src/library_pthread.js

```diff
@@ -188,7 +188,7 @@ var LibraryPThread = {
       // worker pool as an unused worker.
       worker.pthread_ptr = 0;
 
-#if ENVIRONMENT_MAY_BE_NODE && PROXY_TO_PTHREAD
```

Comment:
Could you clarify why this change is necessary? AFAIK, all threads should already be weakly referenced by this point, this should only be required for the "main" thread in `PROXY_TO_PTHREAD` mode, see:
https://github.com/emscripten-core/emscripten/pull/19073#discussion_r1172276555

Reply:
Reverted this part